src/vs/workbench/browser/parts/editor/editorGroupView.ts

```diff
@@ -956,6 +979,58 @@ export class EditorGroupView extends Themable implements IEditorGroupView {
 		return this.model.isActive(editor);
 	}
 
+	async selectEditor(editor: EditorInput, active?: boolean): Promise<void> {
+		if (active) {
+			await this.doOpenEditor(editor, { activation: EditorActivation.ACTIVATE }, { selected: true });
+		} else {
+			this.model.selectEditor(editor, active);
+		}
+	}
+
+	async selectEditors(editors: EditorInput[], activeEditor?: EditorInput): Promise<void> {
+		for (const editor of editors) {
+			await this.selectEditor(editor, editor === activeEditor);
+		}
+	}
+
+	async unSelectEditor(editor: EditorInput): Promise<void> {
+		await this.unSelectEditors([editor]);
+	}
+
+	async unSelectEditors(editors: EditorInput[]): Promise<void> {
+		// Check if the active editor is unselected
+		const unselectingActiveEditor = !!editors.find(editor => this.model.isActive(editor));
+		if (unselectingActiveEditor) {
+			editors = editors.filter(editor => !this.model.isActive(editor));
+		}
+
+		// Unselect all none active editors
+		for (const editor of editors) {
+			this.model.unselectEditor(editor);
+		}
+
+		// if the active editor is unselected, make another selected editor active
+		if (unselectingActiveEditor) {
+			// do not allow to unselect the active editor if it is the last selected editor
+			if (this.selectedEditors.length === 1) {
+				console.warn('Cannot unselect the last selected editor of a group');
```

Comment:
`ILogService` please, and tbh do we need to log this at all? We could return a `boolean` to indicate success of the operation.